src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -57,7 +56,10 @@ static JsonTypeInfo CreateJsonTypeInfo(Type type, JsonSerializerOptions options)
                 }
                 catch (TargetInvocationException ex)
                 {
-                    throw ex.InnerException!;
+                    // Some of the validation is done during construction (i.e. validity of JsonConverter, inner types etc.)
+                    // therefore we need to unwrap TargetInvocationException for better user experience
+                    ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
+                    throw ex.InnerException;
```

Comment:
Don't need to emit a throw opcode here, since the previous method is guaranteed to throw.
```suggestion
                    return null!;
```

Reply:
I'll fix in the next PR